pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -133,10 +135,19 @@ func buildEnvoyLbEndpoint(uid string, family model.AddressFamily, address string
 	// Do not remove: mixerfilter depends on this logic.
 	ep.Metadata = endpointMetadata(uid, network)
 
+	// TODO GregHanson are map keys configured properly
+	if mTLSReady {
```

Comment:
My understanding is that we already have `security.istio.io/mtlsReady` auto populates to envoy endpoint metadata?

Reply:
I misunderstood.  I did not realize that envoy was supporting custom labels, based on the original docs for the transport_socket stuff https://github.com/envoyproxy/envoy/pull/8100/files#diff-1d688ec4c3acf482782efa562b0142c0, I assumed the `enabledMTLS` labels were special labels defined in envoy

Reply:
@diemtvu should we check authn policy is not plain text here? I know it's kinda of annoying for now...